x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/RollupStep.java

```diff
@@ -45,9 +53,13 @@ public void performAction(
         ClusterStateObserver observer,
         ActionListener<Void> listener
     ) {
+        LifecycleExecutionState lifecycleState = indexMetadata.getLifecycleExecutionState();
+        if (lifecycleState.lifecycleDate() == null) {
+            throw new IllegalStateException("source index [" + indexMetadata.getIndex().getName() + "] is missing lifecycle date");
+        }
```

Comment:
Out of curiosity, why do we need to ensure that the lifecycle date is set here?

Reply:
I don't have a good answer for this. I only copied it from the [`ShrinkStep`](https://github.com/elastic/elasticsearch/blob/be7c7415627377a1b795400fb8dfcc6cbdf0e322/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ShrinkStep.java#L62-L65) thinking this may be needed later since`ShrinkStep` and `RollupStep` are pretty similar in their operation.

Do you think I should remove this check?

Reply:
No I don't think it's necessarily a bad check, I was just curious where it came from.